scripts/pre-commit

```diff
@@ -27,14 +27,64 @@ import re
 import subprocess
 import sys
 
-year = str(datetime.datetime.now().year)
-copyr = re.compile(r"(?P<comment>(#|//))\s*Copyright \(C\) (?P<years>[0-9,\s]+)(?P<institution>[\w\s]*)")
+year = datetime.datetime.now().year
+copyr = re.compile(r"(?P<comment>(#|//))\s*Copyright \(C\) (?P<years>[0-9,\s\-]+)(?P<institution>[\w\s]*)")
 # Only apply this to C or Python files since sherpa does not use
 # copyright statements in others files, also, do not change in extern
 # directory.
 ftype = re.compile(r"^((?!extern).)*(hh|cc|c|h|py)$")
 changed = 0
 
+def split_years(years: str) -> list[int]:
+    '''Split a string of years into a list of integers
+
+    This expands ranges of years, e.g. "2010-2012" becomes
+    [2010, 2011, 2012]
+
+    Parameters
+    ----------
+    years : str
+        A comma-separated list of years, with ranges separated by a dash.
+    '''
+    list_years = []
+    for y in years.split(','):
+        if '-' in y:
+            start, end = y.split('-')
+            list_years.extend(range(int(start), int(end) + 1))
+        else:
+            list_years.append(int(y))
+    return list_years
+
+
+def combine_years(list_years: list[int]) -> str:
+    '''Combine a list of years into a string
+
+    This combines consecutive years into ranges, e.g. [2010, 2011, 2012]
```

Comment:
It would be nice to update the code to support duplicates in the input date list (covert the list to be unique). ie... 

    >>> ll=[2011,2012,2010,2011,2016,2012,2015]
    >>> combine_years(ll)
   '2010-2011, 2011-2012, 2012, 2015-2016'

But I would expect the output to look like:

>>> combine_years(list(set(ll)))
'2010-2012, 2015-2016'

Reply:
Since I know that we have a well-kept codebase with careful editing of the list of years in the copyright notice in every file, I was absolutely certain that there can never be an input that is unsorted or contains duplicate entries ;-)